Formula/s-nail.rb

```diff
@@ -0,0 +1,60 @@
+class SNail < Formula
+  desc "Fork of Heirloom mailx."
+  homepage "https://www.sdaoden.eu/code.html"
+  url "https://www.sdaoden.eu/downloads/s-nail-14.9.3.tar.gz"
+  sha256 "9048abe94c8b732ddefcd70b9f6052da16977de76b07598790e4c763e97f911d"
+
+  # Define options
+  option "with-dotlock", "Include support for Dovecot's dotlocking"
+  option "with-spamc", "Includes a client for Spamassasin"
+
+  # Formula depends on OpenSSL and IDNA
+  depends_on "openssl"
+  depends_on "libidn"
+
+  def install
+    args = %W[
+      OPT_AUTOCC=no
+      CC=clang
+      cc_maxtopt=1
+      OPT_NOMEMBDBG=1
+      C_INCLUDE_PATH=#{Formula["openssl"].opt_include}
+      LDFLAGS=-L#{Formula["openssl"].opt_lib}
+      VAL_PREFIX=#{prefix}
+    ]
+
+    # Exclude dotlocking if not defined as option
+    args << "OPT_DOTLOCK=no" if build.without?("dotlock")
+    args << "VAL_PRIVSEP_USER=#{ENV["USER"]}" if build.with?("dotlock")
+
+    # Add optional SpamC
+    args << "OPT_SPAM_SPAMC=yes" if build.with?("spamc")
+
+    # Somehow make doesn't like a file called INSTALL being present.
+    # File is included in the .PHONY variable in the makefile but nonetheless.
+    # Install will fail if not renamed/deleted.
+    mv("INSTALL", "INSTALL.txt")
+    # Configure using above defined args and build
+    system "make", *args, "config"
+    # Build
+    system "make", "build"
+    # Install
+    system "make", "install"
+  end
+
+  def caveats
+    s = ""
+    sdotlock = <<-EOS.undent
+      To make s-nail's dotlocking work you need to make some adjustments:
+
+      sudo chown root #{opt_libexec}/s-nail-privsep
+      sudo chmod u+s #{opt_libexec}/s-nail-privsep
```

Comment:
As mentioned before, we cannot recommend people setuid in Homebrew caveats as a matter of policy.